cvat-ui/src/components/move-task-modal/move-task-modal.tsx

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import './styles.scss';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Modal from 'antd/lib/modal';
+import { Row, Col } from 'antd/lib/grid';
+import Divider from 'antd/lib/divider';
+import notification from 'antd/lib/notification';
+import Tooltip from 'antd/lib/tooltip';
+import { QuestionCircleFilled } from '@ant-design/icons';
+
+import ProjectSearch from 'components/create-task-page/project-search-field';
+import { CombinedState } from 'reducers/interfaces';
+import { closeMoveTaskModal, moveTaskToProjectAsync } from 'actions/tasks-actions';
+import getCore from 'cvat-core-wrapper';
+import LabelMapperItem, { LabelMapperItemValue } from './label-mapper-item';
+
+const core = getCore();
+
+export default function MoveTaskModal(): JSX.Element {
+    const visible = useSelector((state: CombinedState) => state.tasks.moveTask.modalVisible);
+    const task = useSelector((state: CombinedState) => {
+        const [taskInstance] = state.tasks.current.filter((_task) => _task.instance.id === state.tasks.moveTask.taskId);
+        return taskInstance?.instance;
+    });
+    const taskUpdating = useSelector((state: CombinedState) => state.tasks.updating);
+    const dispatch = useDispatch();
+
+    const [projectId, setProjectId] = useState<number | null>(null);
+    const [project, setProject] = useState<any>(null);
+    const [values, setValues] = useState<{ [key: string]: LabelMapperItemValue }>({});
+
+    const initValues = (): void => {
+        if (task) {
+            const labelValues: { [key: string]: LabelMapperItemValue } = {};
+            task.labels.forEach((label: any) => {
+                labelValues[label.id] = {
+                    labelId: label.id,
+                    newLabelName: null,
+                    clearAtrributes: true,
+                };
+            });
+            setValues(labelValues);
+        }
+    };
+
+    const onCancel = (): void => {
+        dispatch(closeMoveTaskModal());
+        initValues();
+        setProject(null);
+        setProjectId(null);
+    };
+
+    const submitMove = async (): Promise<void> => {
+        if (!projectId) {
+            notification.error({
+                message: 'Project not selected',
+            });
+            return;
+        }
+        if (!Object.values(values).every((_value) => _value.newLabelName !== null)) {
```

Comment:
Object.values(values) looks a bit confusing

Reply:
Type of the `value` variable is typed before as an object. So what is exactly look confusing?

Reply:
This piece of code: ``Object.values(values)``. 
Probably ``values`` is not the best name for {key: value} data structure.